File: src/main/java/com/github/unchama/seichiassist/database/manipulators/DonateDataManipulator.java
package com.github.unchama.seichiassist.database.manipulators;

import com.github.unchama.seichiassist.ActiveSkillPremiumEffect;
import com.github.unchama.seichiassist.data.PlayerData;
import com.github.unchama.seichiassist.database.DatabaseConstants;
import com.github.unchama.seichiassist.database.DatabaseGateway;
import com.github.unchama.util.ActionStatus;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Arrays;
import java.util.List;

public class DonateDataManipulator {
    private final DatabaseGateway gateway;

    public DonateDataManipulator(DatabaseGateway gateway) {
        this.gateway = gateway;
    }

    private String getTableReference() {
        return gateway.databaseName + "." + DatabaseConstants.DONATEDATA_TABLENAME;
    }

    public ActionStatus addPremiumEffectBuy(PlayerData playerdata,
                                            ActiveSkillPremiumEffect effect) {
        String command = "insert into " + getTableReference()
                + " (playername,playeruuid,effectnum,effectname,usepoint,date) "
                + "value("
                + "'" + playerdata.name + "',"
                + "'" + playerdata.uuid.toString() + "',"
                + effect.getNum() + ","
                + "'" + effect.getsqlName() + "',"
                + effect.getUsePoint() + ","
                + "cast( now() as datetime )"
                + ")";

        return gateway.executeUpdate(command);
    }

    public ActionStatus addDonate(String name, int point) {
        String command = "insert into " + getTableReference()
                + " (playername,getpoint,date) "
                + "value("
                + "'" + name + "',"
                + point + ","
                + "cast( now() as datetime )"
                + ")";
        return gateway.executeUpdate(command);
    }

    public boolean loadDonateData(PlayerData playerdata, Inventory inventory) {
        ItemStack itemstack;
        ItemMeta itemmeta;
        Material material;
        List<String> lore;
        int count = 0;
        ActiveSkillPremiumEffect[] effect = ActiveSkillPremiumEffect.values();

        String command = "select * from " + getTableReference() + " where playername = '" + playerdata.name + "'";
        try (ResultSet lrs = gateway.executeQuery(command)){
            while (lrs.next()) {
                //ポイント購入の処理
                if(lrs.getInt("getpoint")>0){
                    itemstack = new ItemStack(Material.DIAMOND);
                    itemmeta = Bukkit.getItemFactory().getItemMeta(Material.DIAMOND);
                    itemmeta.setDisplayName(ChatColor.AQUA + "" + ChatColor.UNDERLINE + "" + ChatColor.BOLD + "寄付");
                    lore = Arrays.asList(ChatColor.RESET + "" +  ChatColor.GREEN + "" + "金額：" + lrs.getInt("getpoint")*100,
                            ChatColor.RESET + "" +  ChatColor.GREEN + "" + "プレミアムエフェクトポイント：+" + lrs.getInt("getpoint"),
                            ChatColor.RESET + "" +  ChatColor.GREEN + "" + "日時：" + lrs.getString("date")
                    );
                    itemmeta.setLore(lore);
                    itemstack.setItemMeta(itemmeta);
                    inventory.setItem(count,itemstack);
                }else if(lrs.getInt("usepoint")>0){
                    int num = lrs.getInt("effectnum")-1;
                    material = effect[num].getMaterial();
                    itemstack = new ItemStack(material);
                    itemmeta = Bukkit.getItemFactory().getItemMeta(material);
                    itemmeta.setDisplayName(ChatColor.RESET + "" +  ChatColor.YELLOW + "購入エフェクト：" + effect[num].getName());
                    lore = Arrays.asList(ChatColor.RESET + "" +  ChatColor.GOLD + "" + "プレミアムエフェクトポイント： -" + lrs.getInt("usepoint"),
                            ChatColor.RESET + "" +  ChatColor.GOLD + "" + "日時：" + lrs.getString("date")
                    );
                    itemmeta.setLore(lore);
                    itemstack.setItemMeta(itemmeta);
                    inventory.setItem(count,itemstack);
                }
                count ++;
            }
        } catch (SQLException e) {
            java.lang.System.out.println("sqlクエリの実行に失敗しました。以下にエラーを表示します");
            e.printStackTrace();
            return false;
        }
        return true;
    }
}

File: src/main/java/com/github/unchama/seichiassist/database/manipulators/DonateDataManipulator.kt
package com.github.unchama.seichiassist.database.manipulators

import com.github.unchama.seichiassist.ActiveSkillPremiumEffect
import com.github.unchama.seichiassist.data.PlayerData
import com.github.unchama.seichiassist.database.DatabaseConstants
import com.github.unchama.seichiassist.database.DatabaseGateway
import com.github.unchama.seichiassist.task.recordIteration
import com.github.unchama.util.ActionStatus
import com.github.unchama.util.collection.ImmutableListFactory
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.Material
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.meta.ItemMeta

import java.sql.SQLException

class DonateDataManipulator(private val gateway: DatabaseGateway) {

  private val tableReference: String
    get() = gateway.databaseName + "." + DatabaseConstants.DONATEDATA_TABLENAME

  fun addPremiumEffectBuy(playerdata: PlayerData,
                          effect: ActiveSkillPremiumEffect): ActionStatus {
    val command = ("insert into " + tableReference
        + " (playername,playeruuid,effectnum,effectname,usepoint,date) "
        + "value("
        + "'" + playerdata.name + "',"
        + "'" + playerdata.uuid.toString() + "',"
        + effect.num + ","
        + "'" + effect.getsqlName() + "',"
        + effect.usePoint + ","
        + "cast( now() as datetime )"
        + ")")

    return gateway.executeUpdate(command)
  }

  fun addDonate(name: String, point: Int): ActionStatus {
    val command = ("insert into " + tableReference
        + " (playername,getpoint,date) "
        + "value("
        + "'" + name + "',"
        + point + ","
        + "cast( now() as datetime )"
        + ")")
    return gateway.executeUpdate(command)
  }

  fun loadDonateData(playerdata: PlayerData, inventory: Inventory): Boolean {
    var itemstack: ItemStack
    var itemmeta: ItemMeta
    var material: Material
    var lore2: List<String>
    var count = 0
    val effect = ActiveSkillPremiumEffect.values()

    val command = "select * from " + tableReference + " where playername = '" + playerdata.name + "'"
    try {
      gateway.executeQuery(command).recordIteration {
        val lrs = this
        //ポイント購入の処理
        val getPoint = lrs.getInt("getpoint")
        val usePoint = lrs.getInt("usepoint")
        if (getPoint > 0) {
          itemstack = ItemStack(Material.DIAMOND)
          lore2 = listOf(ChatColor.RESET.toString() + "" + ChatColor.GREEN + "" + "金額：" + getPoint * 100,
              "" + ChatColor.RESET + ChatColor.GREEN + "プレミアムエフェクトポイント：+" + getPoint,
              "" + ChatColor.RESET + ChatColor.GREEN + "日時：" + lrs.getString("date")
          )
          itemstack.itemMeta = Bukkit.getItemFactory().getItemMeta(Material.DIAMOND).apply {
            displayName = "" + ChatColor.AQUA + ChatColor.UNDERLINE + "" + ChatColor.BOLD + "寄付"
            lore = lore2
          }
          inventory.setItem(count, itemstack)
        } else if (usePoint > 0) {
          val num = lrs.getInt("effectnum") - 1
          material = effect[num].material
          itemstack = ItemStack(material)

          lore2 = listOf("" + ChatColor.RESET + ChatColor.GOLD + "プレミアムエフェクトポイント： -" + usePoint,
              "" + ChatColor.RESET + ChatColor.GOLD + "日時：" + lrs.getString("date")
          )
          itemstack.itemMeta = Bukkit.getItemFactory().getItemMeta(material).apply {
            displayName = "" + ChatColor.RESET.toString() + ChatColor.YELLOW + "購入エフェクト：" + effect[num].desc
            lore = lore2
          }
          inventory.setItem(count, itemstack)
        }
        count++
      }
    } catch (e: SQLException) {
      println("sqlクエリの実行に失敗しました。以下にエラーを表示します")
      e.printStackTrace()
      return false
    }

    return true
  }
}
